Marketplace: Add Tracks events to My Subscriptions page

raicem opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
Adds Tracks events for actions taken in the My Subscription page. 
- marketplace_product_install_button_clicked
- marketplace_product_installed
- marketplace_product_install_failed
- marketplace_renew_button_clicked
- marketplace_subscribe_button_clicked
- marketplace_product_update_button_clicked
- marketplace_product_updated
- marketplace_product_update_failed
- marketplace_product_connect_button_clicked

### How to test the changes in this Pull Request:

- Checkout this branch and build assets
- Connect to the `customerwoocommercedotcom` Woo.com account
- Go to WooCommerce > Settings > Advanced > WooCommerce.com and make sure Allow usage of WooCommerce to be tracked option is checked
-  Open JS console and run localStorage.setItem( 'debug', 'wc-admin:*' );. This makes tracks logs show up in console. While in development, tracks events are not sent up to the server.
- Click **Install** on any one of the products
    - Once clicked you should get `marketplace_product_install_button_clicked`
    - Once install finishes you should get `marketplace_product_installed`
    - The props are `product_zip_slug`, `product_id` and `product_current_version`.
-  Find **Google Product Feed** and click **Renew**
    - When clicked we're actually closing this page to go to a new one, the queue system will take place and in the console it'll say _Processing items in queue_ and it'll go to WooCommerce.com.
    - From WooCommerce.com, go back to the in app my subscriptions page. The console will say Processing item in queue. You can look at the arguments for this and see the event name.
- Manually install and outdated plugin by uploading a zip. 
    - Once clicked you should get `marketplace_product_update_button_clicked`
    - Once install finishes you should get `marketplace_product_updated`
- When updated, click the **Connect** button. 
    - Once clicked you should get the `marketplace_product_connect_button_clicked`
- Make the subscribe button show up. You can use this snippet below achieve that. 
    - When clicked, we're actually closing this page to go to a new one, the queue system will take place and in the console it'll say _Processing items in queue_ and it'll go to WooCommerce.com.
    - From WooCommerce.com, go back to the in app my subscriptions page. The console will say Processing item in queue. 

**Snippet to make Subscribe button show up**. Add this to the bottom of the `WC_Helper::get_subscription_list_data` method
```php
// Remove product key. Shows subscribe button.
$subscriptions_to_remove_licence = [
	'W00-61a2c077-4dfc-8a1c-5b52bfc47cf0', // Woo Subscriptions
];

foreach ($subscriptions as &$subscription) {
        if ( in_array( $subscription['product_key'], $subscriptions_to_remove_licence, true ) ) {
		$subscription['product_key'] = '';
	}
}
```

raicem committed:
Marketplace: add Tracks events for the subscription page

Added events for:
- marketplace_product_installed
- marketplace_product_install_failed
- marketplace_renew_button_clicked
- marketplace_subscribe_button_clicked
- marketplace_product_updated
- marketplace_product_update_failed

raicem commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx:
```diff
@@ -66,6 +71,10 @@ export default function Install( props: InstallProps ) {
 					stopInstall();
 				} );
+
+				recordEvent( 'marketplace_product_installed', {
```
Not sure about installed and failed events. They are not much about user behavior and may not have a place in the Tracks system. 

Also more difficult to test the failed events. If we decide to keep them, I'll come up with a way

andfinally replied:
I think it's OK to log these events, myself.

andfinally commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/install.tsx:
```diff
@@ -47,6 +48,10 @@ export default function Install( props: InstallProps ) {
 
 	const install = () => {
+		recordEvent( 'marketplace_product_install_button_clicked', {
+			product_slug: props.subscription.product_slug,
```
Personally I like `product_id` for these kind of events: it's less liable to change than slug. On the other hand, maybe we already use `product_slug` elsewhere, in which case it's fine to be consistent. It's certainly easier to understand. 

It might be useful to add the `version` as well.

raicem replied:
Added version information to install and update Tracks logs.

andfinally commented on plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/update.tsx:
```diff
@@ -36,6 +37,10 @@ export default function Update( props: UpdateProps ) {
 
 	function update() {
+		recordEvent( 'marketplace_product_update_button_clicked', {
+			product_slug: props.subscription.product_slug,
```
The `product_slug` is coming out as `undefined` when I click this button on an item I don't have a subscription to. Not sure what access we have to the product info here.

<img width="500" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/031031f8-3011-43f3-99ff-b8432d5a2448">

raicem replied:
Oh of course, I think you are looking at a state where we can't fetch the list of subscriptions from the server. I say that because product icon is also missing. Apparently, slug is empty in that situation. I bet product id will also be empty. Lemme think what I can do about this.

raicem replied:
`zip_slug` and `product_id` is always present, so switched to logging these two.

andfinally reviewed (commented):
Thanks – looks promising! I'm just wondering how to test this

> Manually install and outdated plugin by uploading a zip.
> 
>   Once clicked you should get marketplace_product_update_button_clicked
>    Once install finishes you should get marketplace_product_updated

This account has a subscription to DePay Payments. But when I manually install an older version of that plugin via the plugins page, the subscription doesn't reflect the fact that the plugin is installed. What method are you using for this bit?

We should probably record an event when the user clicks "Subscribe" on this modal, which appears after you click "Update" on a My Subscriptions item you don't have a subscription for.

<img width="386" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/59481469-2c0c-41e6-b3f4-d037ad90ee1f">

raicem committed:
Marketplace: update Tracks properties

We used to reference product slug only. But may change and if the store
is rate limited, that information is missing. This commit switches
using "product_zip_slug" and "product_id".

raicem committed:
Marketplace: add product version info to update and install Tracks events

kdevnel reviewed (approved):
All looks good to me and tested great. Thanks for these! Just to note, the PR description mentions that the only prop is `product_id` at times. This isn't accurate any more. It's an incredibly minor quibble though and doesn't stop this being approved.

Dan-Q reviewed (approved):
Looks like you've covered all the bases. 🤞

andfinally reviewed (approved):
Thanks! Works nicely! 🙌

raicem committed:
Merge branch 'feature/marketplace-subscriptions' into add/wccom-18721-tracks-events-to-marketplace
